### ISSUE TITLE
Use `sccache` when available

### ISSUE THREAD

**Issue by LeSnake04** [nabijaczleweli/cargo-update#221]
[Sccache](https://github.com/mozilla/sccache/) speeds up the install process by reusing existing compiled binaries.

I think it would be useful if this plugin checks whether sccache is installed and use it if available. 

Using it works by setting `RUSTC_WRAPPER=sccache`.

**Comment by nabijaczleweli** [comment]
Sounds like you want to set that in the environment with the appropriate path, then, right? I don't see how cargo-update needs to (or can) be involved?

**Comment by LeSnake04** [comment]
Yes, It works by setting the environment variable on each run. But its easy to forget to do that. And I dont want to do it via the global cargo config, since this would affect personal rust coding projects as well, which doesnt make sense IMO.

It would be nice if i could tell cargo-update to use it. And this could allow less expericenced users to make use of it just by installing it. Though maybe it makes sense to make it opt-in by adding `cargo install-update-config --enable-sccache` and `cargo install-update-config --disable-sccache` to avoid breaking stuff. Once its clear it doesnt break things make it enable when its available, similar to the cargo-binstall integration. 

Also it would help a lot if there was a env vars option. These variable defined via e.g.  `cargo install-update-config --add-env RUSTC_WRAPPER sccache` . This would also be quite useful for other config customizations users only want for updating cargo applications.

**Comment by alpha-tango-kilo** [comment]
@LeSnake04 `cargo install-update` respects your `~/.cargo/config.toml`, and you can configure `sccache` to be used in there like so:

```toml
[build]
rustc-wrapper = "/path/to/sccache"
```
